go/vt/vttablet/endtoend/message_test.go

```diff
@@ -461,3 +422,168 @@ func TestMessageAuto(t *testing.T) {
 		t.Errorf("message received:\n%v, want\n%v", got, want)
 	}
 }
+
+var createMessageTopic1 = `create table vitess_topic_subscriber_1(
+	time_scheduled bigint,
+	id bigint,
+	time_next bigint,
+	epoch bigint,
+	time_created bigint,
+	time_acked bigint,
+	message varchar(128),
+	primary key(time_scheduled, id),
+	unique index id_idx(id),
+	index next_idx(time_next, epoch))
+comment 'vitess_message,vt_topic=test_topic,vt_ack_wait=1,vt_purge_after=3,vt_batch_size=1,vt_cache_size=10,vt_poller_interval=1'`
+
+var createMessageTopic2 = `create table vitess_topic_subscriber_2(
+	time_scheduled bigint,
+	id bigint,
+	time_next bigint,
+	epoch bigint,
+	time_created bigint,
+	time_acked bigint,
+	message varchar(128),
+	primary key(time_scheduled, id),
+	unique index id_idx(id),
+	index next_idx(time_next, epoch))
+comment 'vitess_message,vt_topic=test_topic,vt_ack_wait=1,vt_purge_after=3,vt_batch_size=1,vt_cache_size=10,vt_poller_interval=1'`
+
+// TestMessageTopic tests for the case where id is an auto-inc column.
+func TestMessageTopic(t *testing.T) {
+	ch1 := make(chan *sqltypes.Result)
+	ch2 := make(chan *sqltypes.Result)
+	done := make(chan struct{})
+	client := framework.NewClient()
+	if _, err := client.Execute(createMessageTopic1, nil); err != nil {
+		t.Fatal(err)
+	}
+	defer client.Execute("drop table vitess_topic_subscriber_1", nil)
+
+	if _, err := client.Execute(createMessageTopic2, nil); err != nil {
+		t.Fatal(err)
+	}
+	defer client.Execute("drop table vitess_topic_subscriber_2", nil)
+
+	// Start goroutine to consume message stream.
+	go waitForMessage(t, client, "vitess_topic_subscriber_1", ch1, done)
+	go waitForMessage(t, client, "vitess_topic_subscriber_2", ch1, done)
+	<-ch1
+	<-ch2
```

Comment:
I'm confused by these two receives.

### when both are commented out
```
newQuery: insert into vitess_topic_subscriber_1(id, message) values(1, 'msg1'), (2, 'msg2'), (3, 'msg3')
panic: send on closed channel

goroutine 150 [running]:
vitess.io/vitess/go/vt/vttablet/endtoend.waitForMessage.func1(0xc00052a140, 0x0, 0x0)
	/vt/src/vitess.io/vitess/go/vt/vttablet/endtoend/message_test.go:585 +0x8e
vitess.io/vitess/go/vt/vttablet/tabletserver.(*QueryExecutor).MessageStream.func2(0xc00052a140, 0x0, 0x0)
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/query_executor.go:264 +0x8c
vitess.io/vitess/go/vt/vttablet/tabletserver/messager.(*messageReceiver).Send.func1(0xc0004cc660, 0xc00052a140)
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/messager/message_manager.go:73 +0x34
created by vitess.io/vitess/go/vt/vttablet/tabletserver/messager.(*messageReceiver).Send
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/messager/message_manager.go:72 +0x61
FAIL	vitess.io/vitess/go/vt/vttablet/endtoend	10.093s
```

### blocks and times out when not commented out
```
goroutine 160 [sync.Cond.Wait, 9 minutes]:
sync.runtime_notifyListWait(0xc0004d8aa8, 0x0)
	/usr/local/go/src/runtime/sema.go:510 +0xeb
sync.(*Cond).Wait(0xc0004d8a98)
	/usr/local/go/src/sync/cond.go:56 +0x92
vitess.io/vitess/go/vt/vttablet/tabletserver/messager.(*messageManager).runSend(0xc0004d8a20)
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/messager/message_manager.go:407 +0x594
created by vitess.io/vitess/go/vt/vttablet/tabletserver/messager.(*messageManager).Open
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/messager/message_manager.go:269 +0xc0

goroutine 161 [select]:
vitess.io/vitess/go/timer.(*Timer).run(0xc00047dc80, 0xc000490e80)
	/vt/src/vitess.io/vitess/go/timer/timer.go:98 +0xc3
created by vitess.io/vitess/go/timer.(*Timer).Start
	/vt/src/vitess.io/vitess/go/timer/timer.go:86 +0xa9

goroutine 178 [select]:
vitess.io/vitess/go/timer.(*Timer).run(0xc00047dca0, 0xc000490e90)
	/vt/src/vitess.io/vitess/go/timer/timer.go:98 +0xc3
created by vitess.io/vitess/go/timer.(*Timer).Start
	/vt/src/vitess.io/vitess/go/timer/timer.go:86 +0xa9

goroutine 179 [chan receive, 9 minutes]:
vitess.io/vitess/go/vt/vttablet/tabletserver.(*QueryExecutor).MessageStream(0xc000109ea0, 0xc00047de00, 0x0, 0x0)
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/query_executor.go:269 +0x1d8
vitess.io/vitess/go/vt/vttablet/tabletserver.(*TabletServer).MessageStream.func1(0xe6c960, 0xc000181350, 0xc0004921e0, 0xc000354000, 0xe6c900)
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/tabletserver.go:1307 +0xef
vitess.io/vitess/go/vt/vttablet/tabletserver.(*TabletServer).execRequest(0xc000354000, 0xe6c960, 0xc000181350, 0x0, 0xd621be, 0xd, 0xd5be51, 0x6, 0x0, 0xc00036bb10, ...)
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/tabletserver.go:1502 +0x34b
vitess.io/vitess/go/vt/vttablet/tabletserver.(*TabletServer).MessageStream(0xc000354000, 0xe6c960, 0xc000181350, 0xc00036bb10, 0xd6e8dc, 0x19, 0xc00047de00, 0x1, 0xc00047de00)
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/tabletserver.go:1291 +0x11d
vitess.io/vitess/go/vt/vttablet/endtoend/framework.(*QueryClient).MessageStream(0xc00036bb00, 0xd6e8dc, 0x19, 0xc00047de00, 0x0, 0x0)
	/vt/src/vitess.io/vitess/go/vt/vttablet/endtoend/framework/client.go:235 +0x67
vitess.io/vitess/go/vt/vttablet/endtoend.waitForMessage(0xc000205a00, 0xc00036bb00, 0xd6e8dc, 0x19, 0xc000350d80, 0xc000350e40)
	/vt/src/vitess.io/vitess/go/vt/vttablet/endtoend/message_test.go:579 +0x86
created by vitess.io/vitess/go/vt/vttablet/endtoend.TestMessageTopic
	/vt/src/vitess.io/vitess/go/vt/vttablet/endtoend/message_test.go:469 +0x2c8

goroutine 183 [chan receive, 9 minutes]:
vitess.io/vitess/go/vt/vttablet/tabletserver.(*QueryExecutor).MessageStream(0xc000109f80, 0xc00047dee0, 0x0, 0x0)
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/query_executor.go:269 +0x1d8
vitess.io/vitess/go/vt/vttablet/tabletserver.(*TabletServer).MessageStream.func1(0xe6c960, 0xc000181350, 0xc0004922d0, 0xc000354000, 0xe6c900)
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/tabletserver.go:1307 +0xef
vitess.io/vitess/go/vt/vttablet/tabletserver.(*TabletServer).execRequest(0xc000354000, 0xe6c960, 0xc000181350, 0x0, 0xd621be, 0xd, 0xd5be51, 0x6, 0x0, 0xc00036bb10, ...)
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/tabletserver.go:1502 +0x34b
vitess.io/vitess/go/vt/vttablet/tabletserver.(*TabletServer).MessageStream(0xc000354000, 0xe6c960, 0xc000181350, 0xc00036bb10, 0xd6e8f5, 0x19, 0xc00047dee0, 0x1623f01, 0xc00047dee0)
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/tabletserver.go:1291 +0x11d
vitess.io/vitess/go/vt/vttablet/endtoend/framework.(*QueryClient).MessageStream(0xc00036bb00, 0xd6e8f5, 0x19, 0xc00047dee0, 0x0, 0x0)
	/vt/src/vitess.io/vitess/go/vt/vttablet/endtoend/framework/client.go:235 +0x67
vitess.io/vitess/go/vt/vttablet/endtoend.waitForMessage(0xc000205a00, 0xc00036bb00, 0xd6e8f5, 0x19, 0xc000350d80, 0xc000350e40)
	/vt/src/vitess.io/vitess/go/vt/vttablet/endtoend/message_test.go:579 +0x86
created by vitess.io/vitess/go/vt/vttablet/endtoend.TestMessageTopic
	/vt/src/vitess.io/vitess/go/vt/vttablet/endtoend/message_test.go:472 +0x343

goroutine 181 [chan receive, 9 minutes]:
vitess.io/vitess/go/vt/vttablet/tabletserver/messager.(*messageManager).Subscribe.func1(0xc000351980, 0xc000391d40, 0xc0004c0f00)
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/messager/message_manager.go:319 +0x34
created by vitess.io/vitess/go/vt/vttablet/tabletserver/messager.(*messageManager).Subscribe
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/messager/message_manager.go:318 +0x231

goroutine 184 [select, 9 minutes]:
vitess.io/vitess/go/vt/vttablet/tabletserver/messager.(*messageReceiver).Send(0xc0004c0f30, 0xc0004caa50, 0xc0004d8a20, 0x1623fa8)
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/messager/message_manager.go:75 +0x101
vitess.io/vitess/go/vt/vttablet/tabletserver/messager.(*messageManager).send(0xc0004d8a20, 0xc000490f60, 0xc0004caa50)
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/messager/message_manager.go:488 +0x181
created by vitess.io/vitess/go/vt/vttablet/tabletserver/messager.(*messageManager).Subscribe
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/messager/message_manager.go:315 +0x1f8

goroutine 185 [chan receive, 9 minutes]:
vitess.io/vitess/go/vt/vttablet/tabletserver/messager.(*messageManager).Subscribe.func1(0xc0003519e0, 0xc0004d8a20, 0xc0004c0f30)
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/messager/message_manager.go:319 +0x34
created by vitess.io/vitess/go/vt/vttablet/tabletserver/messager.(*messageManager).Subscribe
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/messager/message_manager.go:318 +0x231

goroutine 186 [chan send, 9 minutes]:
vitess.io/vitess/go/vt/vttablet/endtoend.waitForMessage.func1(0xc0004caa50, 0x0, 0x0)
	/vt/src/vitess.io/vitess/go/vt/vttablet/endtoend/message_test.go:585 +0x8e
vitess.io/vitess/go/vt/vttablet/tabletserver.(*QueryExecutor).MessageStream.func2(0xc0004caa50, 0xc000320d70, 0xe6c960)
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/query_executor.go:264 +0x8c
vitess.io/vitess/go/vt/vttablet/tabletserver/messager.(*messageReceiver).Send.func1(0xc0004c0f30, 0xc0004caa50)
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/messager/message_manager.go:73 +0x34
created by vitess.io/vitess/go/vt/vttablet/tabletserver/messager.(*messageReceiver).Send
	/vt/src/vitess.io/vitess/go/vt/vttablet/tabletserver/messager/message_manager.go:72 +0x61
```

Reply:
Stupid mistake, I was receiving from `ch1` twice.